documentation/tutorials/!tutorial-template.md

```diff
@@ -0,0 +1,137 @@
+---
+title: {Tutorial Title}
+---
+<!-- template preamble -->
+
+## Template preamble (remove!)
+
+Tutorials are learning-oriented and targeted at beginners or experts to help
+them learn a skill through hands-on experience. A tutorial page shows how to
+accomplish a goal that is larger than a single task, following a single, carefully
```

Comment:
I'm not sure I understand how to scope out a single task vs tutorial - perhaps an example would be helpful

Reply:
**What is a task - the novel**
First concrete examples: 
- A task: add a gateway service
- Not a task: get fully set up as a new API provider (create a gateway service, protect it, share on directory) - too complex, should be broken into tasks. Or covered in a tutorial if we wanted to show ONE path.
- Not a task: Publish configuration using `gwa apply` - too small and isolated, not meaningful

These examples could be added here and/or on tutorial.

Tasks are covered by How-to Guides. Tasks help users get something done. But still, what is a task? And looking at the how-to template, each task is made up of sub-tasks!? 

Going back to Diataxis' food analogy - how-to guides are like recipes in a cook book. Generally, a recipe is for a dish - something you could actually serve (can think, task = solution to a problem). However, sometimes you'll see a recipe for just a dressing or a sauce (enchilada sauce let's say). I think this convention is done because the sauce is reusable in different contexts or it might be too complex unto itself to include in the original (enchilada) recipe. Meanwhile, how to melt butter is not a recipe because it's too simple.

Another way to think about it is: 1) sub-tasks (steps within a task) should not be big and complex, and 2) tasks are meaningful in isolation. 

What I put in the how-to template is also helpful here:
> Only include h3 headings if the sub-task is big and complex and demands splitting.
> Consider if the sub-task deserves it's own how-to page.

It's all a little philosphical so if you have any suggestions on how to put this more concisely in the template preambles, then by all means!

Reply:
this might require some more team discussion. For me - I understand the recipe analogy but because I don't have the technical knowledge I'm not sure how this translates into the work that APS does. If it is clear to devs where the distinction is then that's fine. @rustyjux

Reply:
I'm trying to avoid making the templates overly long and verbose and there are a lot of outside resources to support on the topic. Discussing with James, we expect with time (and discussion when conflicts or uncertainty comes up) we will come to a better understanding of useful examples and boundaries within our context as time goes on and then we can add a short statement or two to the templates to provide guidance.
Marking as resolved.